plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AntlrContentAssistGrammarGenerator.xtend

```diff
@@ -44,7 +44,7 @@ class AntlrContentAssistGrammarGenerator extends AbstractAntlrGrammarWithActions
 				
 				{
 					�FOR kw: allKeywords.sort.sortBy[-length]�
-						tokenNameToValue.put("�keywordHelper.getRuleName(kw)�", "'�kw.toJavaIdentifier(false).replaceAll("\\\\\\$", "\\\\u0024")�'");
+						tokenNameToValue.put("�keywordHelper.getRuleName(kw)�", "'�kw.toStringInAntlrAction.replaceAll("\\\\\\$", "\\\\u0024")�'");
```

Comment:
The replace pattern reads wrong. Xpand did not offer a plain replace, but Java does: We should replace $ by its unicode equivalent